client/src/components/PacmanGame/AnimateEntity.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import { Circle, Image } from 'react-konva';
+import PropTypes from 'prop-types';
+import { entitiesAnimationDurationInSecond, ghostImages, colorCode } from './constants';
+import { locationOnCanvas } from './gameCore';
+
+
+class AnimateEntity extends Component {
+  componentDidMount() {
+    this.animate(this.node, this.center);
+  }
+
+  componentDidUpdate() {
+    this.animate(this.node, this.center);
+  }
+
+  killerCell = ({
+    entity, gridSize, ghostIndex, gridX, gridY,
+  }) => {
+    const image = new window.Image(gridSize, gridSize);
+    image.src = (entity === 'scatterGhost') ? ghostImages[4] : ghostImages[ghostIndex];
+    return (
+      <Image
+        ref={(node) => {
```

Comment:
The use of ref is necessary?

Reply:
Absolutely. No way to do it any other way. 
By react-konva creator: [Getting reference to Konva objects](https://github.com/konvajs/react-konva#getting-reference-to-konva-objects)